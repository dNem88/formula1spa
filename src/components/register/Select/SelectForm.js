import React from 'react'
import {Link, useLocation} from 'react-router-dom'
import styles from './selectForm.module.css'

function SelectForm(props) {
    const location = useLocation()
   
    
    if (props.location === 'auth') {
        return (
        <div className={styles.container}>
            <Link className={location.pathname.includes('/auth/login') ? styles['active-link'] : ''} to={'/auth/login'}>Sign in</Link>
            <Link className={location.pathname.includes('/auth/register') ? styles['active-link'] : ''} to={'/auth/register'}>Register</Link>
        </div>
    )
    }
    if (props.location === 'profile') {
        return (
        <div className={styles.container}>
            <Link className={styles['active-link']} to={'/auth/profile'}>Profile</Link>
        </div>
    )
    }
    if (props.location === 'standings') {
        return (
        <div className={styles.container}>
            <Link className={location.pathname.includes('/standings/drivers') ? styles['active-link'] : ''} to={'/standings/drivers'}>Drivers</Link>
            <Link className={location.pathname.includes('/standings/constructors') ? styles['active-link'] : ''} to={'/standings/constructors'}>Constructors</Link>
        </div>
    )
    }
    
}

export default SelectForm