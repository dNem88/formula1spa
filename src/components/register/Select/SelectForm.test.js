import React from 'react'
import renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'
import SelectForm from './SelectForm'

describe('Testing SelectForm Component', () => {
    test('snapshot match', () => {
        let Select = renderer.create(
            <MemoryRouter>
                <SelectForm/>
            </MemoryRouter>
        ).toJSON()
        expect(Select).toMatchSnapshot()
        expect(Select.type).toBe('div')
        expect(Select.children.length).toBe(2)
        expect(Select.children[0].type).toBe('a')
        expect(Select.children[1].type).toBe('a')
        expect(Select.children[0].props.href).toBeTruthy()
        expect(Select.children[1].props.href).toBeTruthy()
        expect(Select.children[0].props.className).toBeFalsy()
        expect(Select.children[1].props.className).toBeFalsy()
        expect(Select.children[0].children[0]).toBe('Sign in')
        expect(Select.children[1].children[0]).toBe('Register')
    })
})