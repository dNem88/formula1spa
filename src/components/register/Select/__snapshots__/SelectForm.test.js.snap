// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SelectForm Component snapshot match 1`] = `
<div
  className="container"
>
  <a
    className=""
    href="/auth/login"
    onClick={[Function]}
  >
    Sign in
  </a>
  <a
    className=""
    href="/auth/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
